File: source/net/auoeke/cin/lexer/lexeme/IntegerLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class IntegerLexeme extends Lexeme {
    public final long value;

    public IntegerLexeme(int line, int column, long value, SyntaxError error) {
        super(line, column, error);

        this.value = value;
    }

    public IntegerLexeme(int line, int column, long value) {
        this(line, column, value, null);
    }

    @Override public Type type() {
        return Type.INTEGER;
    }

    @Override public String toString() {
        return Long.toString(this.value);
    }
}

File: src/net/auoeke/cin/lexer/lexeme/IntegerLexeme.kt
package net.auoeke.cin.lexer.lexeme

import net.auoeke.extensions.string

class IntegerLexeme(line: Int, column: Int, val value: Long) : Lexeme(line, column) {
    override val type: Type get() = Type.INTEGER

    override fun toString(): String = value.string
}
